fix: avoid ansi escape codes in non-tty and no color output

jahed-snyk opened the pull request:
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

When the terminal doesn't support colour (either NO_COLOR env variable or not TTY as detected by NodeJS), it will remove every ANSI escape code when printing the `help` output. This avoids printing them in plain text and obfuscating the help text.

#### Where should the reviewer start?

Only a few files to check.

#### How should this be manually tested?

Run `snyk help | cat` will print ANSI codes in plain text before this change. After it should print plain unformatted text.

#### Any background context you want to provide?

Originally noticed in some Powershell sessions.

#### What are the relevant tickets?

None.

#### Screenshots

None.

#### Additional questions

`strip-ansi` is a part of `chalk` so it's not added as a direct dependency, but as best practice, maybe it should be as we use it directly?

maxjeffos commented on src/cli/commands/help.ts:
```diff
@@ -1,8 +1,17 @@
 
 const DEFAULT_HELP = 'snyk';
 
+const readHelpFile = (filename: string): string => {
```
Don't think we have a policy on this, but this kind of function definition is not particularly helpful unless it is required (i.e. when using as a closure) and does not appear to be inline with the norm.

jahed-snyk replied:
I've changed it to use plain old functions.

maxjeffos reviewed (changes requested):
The PR description could use some more info, too (like "why"). Is there a reason you didn't use at least part of the PR description template?

jahed-snyk commented:
>Is there a reason you didn't use at least part of the PR description template?

I asked and was told it was fine. I agree though; the why is important. I'll polish this up tomorrow. 👍

JackuB commented:
Ha, my bad, told @jahed-snyk to not fill it :)

jahed-snyk commented:
Updated OP with details now. Also added a question:

>strip-ansi is a part of chalk so it's not added as a direct dependency, but as best practice, maybe it should be?

jahed-snyk commented:
Merged. To answer the remaining question, we already have strip-ansi in the package.json. 🙃
